tests/erlang_tests/test_code_get_object_code.erl

```diff
@@ -0,0 +1,66 @@
+%
+% This file is part of AtomVM.
+%
+% Copyright 2025 Franciszek Kubis <franciszek.kubis@swmansion.com>
+%
+% Licensed under the Apache License, Version 2.0 (the "License");
+% you may not use this file except in compliance with the License.
+% You may obtain a copy of the License at
+%
+%    http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS,
+% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+% See the License for the specific language governing permissions and
+% limitations under the License.
+%
+% SPDX-License-Identifier: Apache-2.0 OR LGPL-2.1-or-later
+%
+
+-module(test_code_get_object_code).
+
+-export([start/0]).
+
+-include("code_load/export_test_module_data.hrl").
+
+start() ->
+    ok = get_object_from_export_test_module(),
+    ok = get_object_from_already_loaded_test_module(),
+    ok = get_object_from_non_existing_module(),
+    ok = get_object_wrong_argument("a string"),
```

Comment:
compiler or typechecking might understand something is wrong here and raise an error. My suggestion is always using some kind of indirection, such as `?MODULE:get_object_wrong_argument`, so the compiler cannot make any kind of clever check.